language-support/java/bindings/src/main/java/com/daml/ledger/javaapi/data/DamlGenMap.java

```diff
@@ -22,15 +21,30 @@ public DamlGenMap(java.util.Map<Value, Value> value) {
         this.value = value;
```

Comment:
Is `DamlGenMap` supposed to be immutable? If it is, we might want to clone the map here (or make this private, and make a public factory method that clones it).

Reply:
It is not suppose to be immutable. At least I copy from `DamlList` where the underlying list is mutable.
@gerolf-da should I make `DamlList`, `DamlMap`, and `DamlGenMap` immutable ?

Reply:
Yes, I was looking at `DamlList` recently in another context and I noticed it there too. Ideally we copy the map (the values themselves should be already immutable).